src/uu/dd/src/dd.rs

```diff
@@ -719,16 +763,18 @@ fn handle_o_direct_write(f: &mut File, buf: &[u8], original_error: io::Error) ->
 
         write_result
     } else {
-        // O_DIRECT wasn't set, return original error
+        // O_DIRECT wasn't set or this is a full block, return original error
         Err(original_error)
     }
 }
 
 /// Stub for non-Linux platforms - just return the original error.
 #[cfg(not(any(target_os = "linux", target_os = "android")))]
+#[allow(dead_code)]
 fn handle_o_direct_write(
```

Comment:
is this stub still needed? I guess its not called an non-linux anymore, right?
e.g. there is already a stub for "write_with_o_direct_handling" which doesn't call this helper anymore